### PR TITLE
core: Fix filters being cut off when nested inside another cacheAsBitmap

### PR DESCRIPTION
**Pull request by Dinnerbone**
This should fix any remaining filter-cutoff issue, such as the glow in #11729 (https://github.com/ruffle-rs/ruffle/issues/11729#issuecomment-1632913648).

### CODE REVIEW

**Comment by waspennator** [comment]
Fixes the glow on the quality text being slightly cutoff on #9621

File: swf/src/types/blur_filter.rs [hunk 1]
```diff
@@ -31,10 +31,10 @@ impl BlurFilter {
         self.num_passes() == 0 || (self.blur_x <= Fixed16::ONE && self.blur_y <= Fixed16::ONE)
     }
 
-    pub fn calculate_dest_rect(&self, source_rect: Rectangle<i32>) -> Rectangle<i32> {
+    pub fn calculate_dest_rect(&self, source_rect: Rectangle<Twips>) -> Rectangle<Twips> {
         let scale = PASS_SCALES[self.num_passes().clamp(1, 15) as usize - 1];
-        let x = (scale * self.blur_x.to_f32()).ceil().max(0.0) as i32;
-        let y = (scale * self.blur_y.to_f32()).ceil().max(0.0) as i32;
+        let x = Twips::from_pixels(scale * self.blur_x.to_f64());
+        let y = Twips::from_pixels(scale * self.blur_y.to_f64());
```
Comment (torokati44) [inline swf/src/types/blur_filter.rs]: Is the `.ceil().max(0.0)` here not necessary anymore? Or was this an accidental change?

**Comment by torokati44** [comment]
Just curious, why the switch to `Twips`?

File: swf/src/types/blur_filter.rs (hunk 1 above)
Comment (Dinnerbone) [reply swf/src/types/blur_filter.rs]: Hmm, max should stay, but ceil is handled by twips->int

**Comment by Dinnerbone** [comment]
Because bounding boxes work in Twips, and that's where this is mostly called now

File: core/src/display_object.rs [hunk 2]
```diff
@@ -1166,17 +1179,29 @@ pub trait TDisplayObject<'gc>:
     /// This differs from the bounds that are exposed to Flash, in two main ways:
     /// - It may be larger if filters are applied which will increase the size of what's shown
     /// - It does not respect scroll rects
-    fn render_bounds_with_transform(&self, matrix: &Matrix) -> Rectangle<Twips> {
+    fn render_bounds_with_transform(
+        &self,
+        matrix: &Matrix,
+        include_own_filters: bool,
+        view_matrix: &Matrix,
+    ) -> Rectangle<Twips> {
```
Comment (torokati44) [inline core/src/display_object.rs:1187]: Function signature changes without doc comment changes are always _sus_, but in this case, I think the same comment still applies and is sufficient.

Comment (Dinnerbone) [reply core/src/display_object.rs:1187]: I wrote that comment just the other day whilst thinking about this change 😛

File: swf/src/types/blur_filter.rs (hunk 1 above)
Comment (Dinnerbone) [reply swf/src/types/blur_filter.rs]: Fixed!

**Review by torokati44 — APPROVED** [review]
Mostly a rubber-stamp, but LGTM! And happy with all the stuff it fixes!

**Comment by torokati44** [comment]
Hmh, I guess a visual test would have been nice... Maybe later?

**Comment by Dinnerbone** [comment]
There was one...? 😅

**Comment by torokati44** [comment]
Right, yes, sorry, my bad.

**Pull request merged**